File: LeetCode/Medium/Container With Most Water/Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        s = 0
        
        while left < right:
            if height[left] > height[right]:
                s = max(s, height[right]*(right-left))
                right -= 1
            else:
                s = max(s, height[left]*(right-left))
                left += 1
        return s

File: LeetCode/Medium/Container With Most Water/Container With Most Water.go
func maxArea(height []int) int {
    left, right := 0, len(height)-1
    maxContainer := 0
    tmpContainer := 0
    for left < right{
        if height[right] > height[left]{
            tmpContainer = height[left]*(right-left)
            left ++
        }else{
            tmpContainer = height[right]*(right-left)
            right --
        }
        maxContainer = Max(maxContainer, tmpContainer)
    }
    return maxContainer
}

func Max (a, b int) int{
    if a>=b{
        return a
    }
    return b
}
